src/formatDate.js

```diff
@@ -7,8 +7,50 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(date, sourceFormat, targetFormat) {
+  // Делаем деструктуризацию кода
+  const [, , , sourceSeparator] = sourceFormat;
+  const [, , , targetSeparator] = targetFormat;
+
+  // Разделение даты по сепаратору
+  const dateParts = date.split(sourceSeparator);
+  const [sourceYear, sourceMonth, sourceDay] = dateParts;
+
+  const formattedDateParts = [];
+
+  // Переформатирование даты в зависимости от исходного формата
+  if (
+    sourceFormat[0] === 'YYYY' &&
+    targetFormat[0] === 'DD' &&
+    targetFormat[1] === 'MM' &&
+    targetFormat[2] === 'YYYY'
+  ) {
+    formattedDateParts.push(sourceDay, sourceMonth, sourceYear);
+  } else if (targetFormat[2] === 'YY' && sourceFormat[2] === 'YYYY') {
+    formattedDateParts.push(
+      dateParts[0],
+      dateParts[1],
+      dateParts[2].slice(2, 4),
+    );
+  } else if (sourceFormat[0] === 'YY' && targetFormat[0] === 'YYYY') {
+    const fullYear = sourceYear < 30 ? `20${sourceYear}` : `19${sourceYear}`;
+
+    formattedDateParts.push(fullYear, sourceMonth, sourceDay);
```

Comment:
The condition correctly converts a two-digit year to a four-digit year, but it assumes that the year is the first element of the sourceFormat. You need to handle the case where the year can be in any position in the sourceFormat array.